packages/makeup-hoist-element/src/index.js

```diff
@@ -0,0 +1,70 @@
+let hoistEl;
+let hoistedElementPlaceholder;
+let containerDiv;
+let bodyChildIndexes = [];
+const BODY_TAG_NAME = 'BODY';
+
+function isHoisted(element) {
+    if (element.parentElement.tagName === BODY_TAG_NAME) {
+        return true;
+    }
+    return false;
+}
+
+function unhoist() {
+    if (hoistEl) {
+        if (containerDiv) {
+            [...containerDiv.childNodes].forEach((child) => {
+                if (!child.src) {
+                    const index = bodyChildIndexes.shift();
+                    if (index > document.body.childNodes.length) {
+                        document.body.appendChild(child);
+                    } else {
+                        document.body.insertBefore(child, document.body.childNodes[index + 1]);
+                    }
+                }
+            });
+            containerDiv.remove();
+            containerDiv = null;
+            bodyChildIndexes = [];
+        }
+
+        if (hoistedElementPlaceholder) {
+            hoistedElementPlaceholder.replaceWith(hoistEl);
+            // let observers know the element is unhoisted
+            hoistedElementPlaceholder = null;
+        }
+        hoistEl = null;
+    }
+    return hoistEl;
+}
+
+function hoist(el) {
+    if (isHoisted(el)) {
+        unhoist();
+    }
+    hoistEl = el;
+
+    hoistedElementPlaceholder = document.createElement('div');
+    hoistEl.parentElement.insertBefore(hoistedElementPlaceholder, hoistEl);
+
+    containerDiv = document.createElement('div');
+    [...document.body.childNodes].forEach((child, index) => {
+        if (!child.src) {
```

Comment:
Is the intention here to filter out script tags? If so, we should create a filter function that filters out array elements based on tagName.

Reply:
Well here's the issue: I needed to keep the script tags relatively in the same place when I `hoist`, and then when I `unhoist` I need to move them back to the same place I started with. That was my objective with this `forEach`.